### ISSUE TITLE
[feature] Reference variables/envs/keys from other packages

### ISSUE THREAD

**Issue by artsiommiksiuk** [moonrepo/moon#1618]
**Is your feature request related to a problem? Please describe.**

E.g. I have a package A and package B. B depends on A to be build. I'd like to reference some files/outputs from A in B, so I don't have to copy/paste same strings in B.

This concept is close by it's utility as Bazel does it's artifacts referencing, where you can reference anything anywhere, given that the proper path in Bazel space specified.

**Describe the solution you'd like**

The way I see this is to have something like openAPI allows referencing

```yaml
# a's moon.yaml

env:
  OUTPUT_DIR: "out/dir"
```

```yaml
# b's moon.yaml
task:
  script: "echo @in(0)"
  inputs:
    - $ref: "A:envs.OUTPUT_DIR"
```

**Describe alternatives you've considered**

1. Wait for .PKL migration?
2. Copy/pasting same output path

**Additional context**

Basically I have a package which produces some artifact as a set of files. This files needs later to be used in another project in the same monorepo, for that I need to know the path where to copy them from.
